presto-docs/src/main/sphinx/security/server.rst

```diff
@@ -0,0 +1,281 @@
+
+Kerberos Principals and Keytab Files
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The Presto coordinator needs a Kerberos principal, as do users who are going to
+connect to the Presto coordinator. You will need to create these users in
+Kerberos using `kadmin
+<http://web.mit.edu/kerberos/krb5-latest/doc/admin/admin_commands/kadmin_local.html>`_.
+
+In addition, the Presto coordinator needs a `keytab file
+<http://web.mit.edu/kerberos/krb5-devel/doc/basic/keytab_def.html>`_. After you create the principal, you can create the keytab file using :command:`kadmin`
+
+.. code-block:: none
+
+  kadmin
+  > addprinc -randkey presto@EXAMPLE.COM
+  > addprinc -randkey presto/presto-coordinator.example.com@EXAMPLE.COM
+  > ktadd -k /etc/presto/presto.keytab presto@EXAMPLE.COM
+  > ktadd -k /etc/presto/presto.keytab presto/presto-coordinator.example.com@EXAMPLE.COM
+
+.. include:: ktadd-note.fragment
+
+.. include:: jce-policy.fragment
+
+.. _server_java_keystore:
+
+Java Keystore File for TLS
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When using Kerberos authentication, access to the Presto coordinator should be
+through HTTPS. The Presto coordinator needs keys to secure the :abbr:`TLS
+(Transport Layer Security)` connection. These keys are generated using
+:command:`keytool` and stored in a Java Keystore file for the Presto
+coordinator.
+
+The alias in the :command:`keytool` command line should match the principal that the
+Presto coordinator will use.
+
+You'll be prompted for the first and last name. Use the Common Name that will
+be used in the certificate. In this case it should be the unqualified hostname
+of the Presto coordinator. In the example, you can see this in the prompt that
+confirms the information is correct.
+
+.. code-block:: none
+
+  keytool -genkeypair -alias presto -keyalg RSA -keystore keystore.jks
+  Enter keystore password:
+  Re-enter new password:
+  What is your first and last name?
+    [Unknown]:  presto-coordinator
+  What is the name of your organizational unit?
+    [Unknown]:
+  What is the name of your organization?
+    [Unknown]:
+  What is the name of your City or Locality?
+    [Unknown]:
+  What is the name of your State or Province?
+    [Unknown]:
+  What is the two-letter country code for this unit?
+    [Unknown]:
+  Is CN=eiger, OU=Unknown, O=Unknown, L=Unknown, ST=Unknown, C=Unknown correct?
+    [no]:  yes
+
+  Enter key password for <presto>
+          (RETURN if same as keystore password):
+
+
+.. _server_access_controller:
+
+Access Controller Implementation
+--------------------------------
+
+Presto separates the concept of the principal who authenticates to the
+coordinator from the username that is responsible for running queries. When
+running the Presto CLI, for example, the Presto username can be specified using
+the --user option.
+
+By default, the Presto coordinator allows any principal to run queries as any
+Presto user. In a secure environment, this is probably not desirable behavior.
```

Comment:
File a task for this.  I had this on my todo list when writing the security code and simply forgot.  My plan was to make the default implementation that requires the username to match the Kerberos principal name.